src/components/main-view/main-view.jsx

```diff
@@ -26,7 +37,22 @@ export const MainView = () => {
 
       setMovies(moviesfromApi);
     });
-  }, []);
+  }, [token]);
+
+  if (!user) {
+    return (
+    <>
+      <LoginView 
+        onLoggedIn={(user, token) => {
```

Comment:
You established a pattern with handleSubmit that you are not passing inline functions, extract this to a `const handleOnLoggedIn`